src/staticanalysis/bot/static_analysis_bot/clang/tidy.py

```diff
@@ -244,6 +244,14 @@ def __init__(self, header_data, revision):
         self.body = None
         self.notes = []
 
+        self.reason = None
+        for c in settings.clang_checkers:
+            name = c['name']
+
+            # Wildcard at the end of check name and also normal matching
+            if name == self.check or (name.endswith('*') and self.check.startswith(name[:-1])):
```

Comment:
You duplicated the code from [settings.is_publishable_check](https://github.com/mozilla/release-services/blob/4625854c25aa6630184751a85325d30de2756c8d/src/staticanalysis/bot/static_analysis_bot/config.py#L97) instead of creating a common function shared between `settings.is_publishable_check` and here, as requested.

Please do the following steps:

1. create a common function in [settings.py](https://github.com/mozilla/release-services/blob/4625854c25aa6630184751a85325d30de2756c8d/src/staticanalysis/bot/static_analysis_bot/config.py#L97), named `get_clang_check`
2. `get_clang_check` must be the only place where this decision code is present
3. `get_clang_check` must return the dict containing all information of the required check description, or `None` if no check is present
4. Update `settings.is_publishable_check` to use `get_clang_check`. Its behaviour must remain the same.
5. Finally use `get_clang_check` here like so:

```python
check = settings.get_clang_check(self.check)
if check is not None:
  self.reason = check.get('reason')
```